src/api/sso/test/row-level-policies.test.js

```diff
@@ -0,0 +1,348 @@
+};
+
+// unregistered user
+const unregisteredUser = {
+  firstName: 'Galileo',
+  lastName: 'Galilei',
+  email: 'user2@example.com',
+  displayName: 'Galileo Galilei',
+  isAdmin: false,
+  isFlagged: false,
+  feeds: ['https://dev.to/galileogalilei'],
+  githubUsername: 'octocat',
+  githubAvatarUrl: 'https://avatars.githubusercontent.com/u/583231?v=4',
+};
+
+// Create 2 Telescope accounts in the database
+const users = [hansLippershey, johannesKepler];
+
+// Create 2 feeds, 1 belongs to a user, 1 does not belong to anyone
+const feedList = [
+  {
+    url: hansLippershey.feeds[0],
+    wikiAuthorName: hansLippershey.displayName,
+    userId: hash(hansLippershey.email),
+  },
+  {
+    url: unregisteredUser.feeds[0],
+    wikiAuthorName: unregisteredUser.displayName,
+  },
+];
+
+describe('Unauthenticated users with public anon_key', () => {
+  beforeAll(() => {
+    supabase = createClient(SUPABASE_URL, ANON_KEY);
+  });
+
+  describe('"telescope_profiles" security policies', () => {
+    beforeEach(() => createTelescopeUsers(users));
+
+    afterEach(() => cleanupTelescopeUsers(users));
+
+    it('should return no users', async () => {
+      const { status, count } = await supabase
+        .from('telescope_profiles')
+        .select('*', { count: 'exact' });
+
+      // the request succeeds but no rows are read
+      expect(status).toBe(200);
+      expect(count).toBe(0);
+    });
+
+    it('should not allow creating new users', async () => {
+      const { status } = await supabase.from('telescope_profiles').insert(
+        {
+          id: hash(unregisteredUser.email),
+          email: unregisteredUser.email,
+          first_name: unregisteredUser.firstName,
+          last_name: unregisteredUser.lastName,
+          display_name: unregisteredUser.displayName,
+          github_username: unregisteredUser.githubUsername,
+          github_avatar_url: unregisteredUser.githubAvatarUrl,
+        },
+        { returning: 'minimal' }
+      );
+
+      // forbidden, users cannot be created on the client
+      expect(status).toBe(403);
+    });
+
+    it('should not allow updating users', async () => {
+      const { status, count } = await supabase
+        .from('telescope_profiles')
+        .update({ display_name: unregisteredUser.displayName })
+        .match({ id: hash(users[0].email) });
+
+      // no rows are returned to be updated, therefore 404
+      expect(status).toBe(404);
+    });
+
+    it('should not allow deleting users', async () => {
+      const { status, count } = await supabase
+        .from('telescope_profiles')
+        .delete({ count: 'exact' })
+        .match({
+          id: hash(users[0].email),
+        });
+
+      // no rows are deleted
+      expect(status).toBe(200);
+      expect(count).toBe(0);
+    });
+  });
+
+  describe('"feeds" security policies', () => {
+    // Create feeds for HansLippershey and an unregistered user
```

Comment:
Should this be:

```js
// Create feeds for HansLippershey, an unregistered user
```

If not, it reads like there should be 2 users here, not one.

Reply:
It's meant to create feeds for 2 users, one is registered, Hanlippershey, and one is not.